qiskit-superstaq/qiskit_superstaq/superstaq_job.py

```diff
@@ -37,9 +45,31 @@ def __eq__(self, other: Any) -> bool:
 
         return self._job_id == other._job_id
 
+    def get_job_id(self) -> str:
+        """Returns the virtual job id for the job."""
+        return self._job_id
+
+    def get_backend(self) -> str:
+        """Returns the job backend used as a string."""
+        self._check_if_stopped()
+        return self._backend
+
     def _wait_for_results(
```

Comment:
Yes we can, but then the timeout cannot be optional (otherwise `wait_for_final_state()` would wait indefinitely I believe) and so we would need to set a default timeout.

Reply:
> otherwise `wait_for_final_state()` would wait indefinitely I believe

i believe this is what we do now if the job never ends up in a final state? but also definitely fine w/ setting a default

another option is to default to the client's timeout, e.g.
```python
timeout = timeout or self._backend._provider._client.max_retry_seconds
```

Reply:
I think it's better to default to the client's timeout, so I've gone with that in 3c9aed3